### ISSUE TITLE
Intermediate points in definition of dragging (for SC2.5.7)

### ISSUE THREAD

**Issue by guyhickling** [w3c/wcag#1195]
In the definition of "dragging movement" in the WCAG 2.2 Glossary what does "without requiring movement through an intermediate point" mean? It is not explained anywhere, and does not appear to have any relevance to the matter of dragging (that I can see). Is this a carry-over from some past discussion or early thoughts in creating this SC?

Also, why should drags through intermediate points be excluded from this SC? I can think of at least one scenario where going through an intermediate point would be a perfectly valid thing to do that should not be excluded. Take a children's game, where they have a row of objects along the top of the page, and they have to drag down through the correct one of a choice of gates along the middle of the page, and leave them in piles at the foot of the page. The gates are  intermediate points, but this drag operation should still be possible using single pointers (e.g. providing up/down/left/right buttons to click or tap on) same as any other drag operation.

So I suggest deleting the phrase "without requiring movement through an intermediate point" from the definition of dragging, or else explain what it means.

**Comment by detlevhfischer** [comment]
@guyhickling You are right that the phrase "without requiring movement through an intermediate point" may be hard to understand without reference to 2.5.1 Pointer Gestures, where the requirement of the movement to go through an intermediate point was a technical expression of the directionality of the gesture - see [Understanding Pointer Gestures](https://w3c.github.io/wcag/understanding/pointer-gestures.html). The phrase was chosen to separate swiping gestures from dragging, which could go in any direction (the element under the pointer just 'latching on' and following the pointer movement). 

> Also, why should drags through intermediate points be excluded from this SC? 

They are not excluded - the phrase just says that in contrast to single pointer gestures, dragging is a movement that does not require (but also, does not exclude) going through an intermediate point.  Maybe this should be made clearer in the definition,

> So I suggest deleting the phrase "without requiring movement through an intermediate point" from the definition of dragging, or else explain what it means.

I think the phrase is useful to separate Dragging from Pointer Gestures (as understood in 2.5.1). As to providing an explanation, the Understanding texts are typically the place where such explanations should be given. I'm not sure that a longish explanation in the glossary would be appropriate. Maybe a cross reference to "single pointer "would help.

**Comment by guyhickling** [comment]
Thank you. If I am understanding you correctly, you are saying that the words "without requiring movement through an intermediate point", in that "dragging movement" glossary definition, are referring to path-based gestures on touch screens - is that correct?

In other words, "dragging movement" does not include path-based gestures. Which in turn means path-based gestures are not covered by this SC (i.e. they don't have to have an alternative single pointer method). 

But that can't be right surely, because the SC is specifically about dragging movements, and the Understanding doc for 2.5.7 says that it does include path-based gestures as well. Under the Benefits section it says: 

> Users who struggle with performing path-based gestures can still operate an interface with a pointer interface.

NB: if path-based gestures are not covered by this SC then there should be another SC for them, as they are just as impossible for people using certain types of pointer equipment as are dragging operations. I think it should include both these two operations.

**Comment by detlevhfischer** [comment]
> I am understanding you correctly, you are saying that the words "without requiring movement through an intermediate point", in that "dragging movement" glossary definition, are referring to path-based gestures on touch screens - is that correct?

yes

> In other words, "dragging movement" does not include path-based gestures. Which in turn means path-based gestures are not covered by this SC (i.e. they don't have to have an alternative single pointer method).

They are covered by the existing SC 2.5.1 Pointer Gestures - the new proposed SC now closes the gap by covering dragging movements.

> the Understanding doc for 2.5.7 says that it does include path-based gestures as well.

The general observation ('users struggle') is true for both dragging and path-based movements. I guess we can update the Understanding to make it clearer that here, the focus is on dragging, not swiping and multipoint gestures.

> NB: if path-based gestures are not covered by this SC then there should be another SC for them

Tha's what 2.5.1 is covers already.

**Comment by detlevhfischer** [comment]
I have created a PR #1208 to change the Understanding > Benefits text (referring now to 'dragging movements' instead of 'path based gestures').

**Comment by guyhickling** [comment]
Thank you, replacing the gestures reference by dragging will greatly improve it. 

While you are doing that, could you please also add some explanation into the Understanding for this SC what the ""without requiring movement through an intermediate point" means, and explain why it is present in the glossary definition of dragging? Most people won't know they have to go to SC2.5.1's Understanding to find the reference, and even if they do it is not clear how it applies to 2.5.7 given that swiping gestures don't apply to dragging, or to this SC.

**Comment by detlevhfischer** [comment]
> could you please also add some explanation into the Understanding for this SC about what the "without requiring movement through an intermediate point" means

I think this would be confusing as the expression is now nowhere present in the [revised Dragging Understanding text](https://raw.githack.com/w3c/wcag/wcag22-pointer-target-spacing/understanding/22/pointer-target-spacing.html).

**Comment by guyhickling** [comment]
Detlev, I was just about to point that out, but you spotted it first. It may help that the Dragging SC is the only SC to refer to that dragging movement definition (which of course comes from the glossary initially). So it can be amended without affecting any other SCs. Since that "intermediate point" phrase refers to path-based gestures on touch screens, and has nothing to do with dragging, that is why I think it doesn't belong in the definition of dragging movement at all. It just causes confusion being there (it certainly confused the stuffing out of me!)

**Comment by alastc** [comment]
PR #1214 updated the definition.

> just realising that the active version of the Dragging Understanding text does contain the phrase under Key terms

The live version has been updated, the phrase doesn't appear now.

I've tagged #1208 as "ready to survey", so we'll get to that in a couple of weeks (ish).

I think that tackles everything in this thread, but please re-open if I've missed something.
